### ISSUE TITLE
user position indicators triggers different bugs (what's that feature correctly called?)

### ISSUE THREAD

**Issue by kolAflash** [nextcloud/text#4100]
The new _"user position indicator"_ feature triggers different bugs.

<details>
<summary>screenshot</summary>

![Screenshot_20230501-222621_Nextcloud](https://user-images.githubusercontent.com/3355089/235526420-7c7984cb-9470-42a7-aaa0-8d97140c8523.png)
</details>

&nbsp;

### Questions, Problems and Bugs

Questions:
- What's the correct name for that feature!? (I just call it _"User position indicator"_ because I don't know any other name for it)
- Can the feature be disabled?
  - It's a nice feature. But for me it's not worth the problems and bugs it causes.
- The indicators seem to go away 30 seconds after opening a document. Can that time span be changed?

&nbsp;

Bugs:
- Sometimes all text following an indicator is deleted. This seems to happen when editing text at the position of the indicator. (couldn't determine the exact circumstances until now) 
- In the browser (Firefox) empty lines containing an indicator can't be clicked.
- Same user is listed multiple times, although he's just logged in from one client. (counted up to five)

&nbsp;

Problems:
- When an user position indicator overlaps text, the text can't be read.
- Sometimes the same indicator seems to be multiple times at the same position. Clicking the first indicator in the screenshot (see above) on Android makes my keyboards auto completion show `kolAflashkolAflashkolAflashkolAflashkolAflash`. (5 times)

&nbsp;

&nbsp;

### Tested platforms

- PC (Firefox-112.0.2)
- Android-10 (Nextcloud App 3.24.2)
- Android-13 (Nextcloud App 3.24.2) 

Server:
Nextcloud: 26.0.1
Text App: 3.7.2
OS: Debian-12 (Apache-2.4.57, php-fpm-8.2.4, mariadb-10.11.2)

**Comment by juliushaertl** [comment]
Thanks for reporting, for the first issue it would be quite helpful to have more detailed steps, maybe also a short screen recording to see when this happens.

> Same user is listed multiple times, although he's just logged in from one client. (counted up to five)

This has been addressed with https://github.com/nextcloud/text/pull/4053 already and will be fixed in the next maintenance release.

**Comment by kolAflash** [comment]
@juliushaertl  
Sure. Here are some, showing a wide range of problems.  
Including the bug which deletes all text following the cursor.

[FP3_Android-10_eOS-1.4.mp4](https://user-images.githubusercontent.com/3355089/236349797-1b748231-1289-4128-83a9-d45a8e062ef3.mp4)

[Pixel4a_Android-13.mp4](https://user-images.githubusercontent.com/3355089/236349937-6e0c44a1-1036-4d22-8cf7-6a1c85fb5b0c.mp4)

One problem I newly discovered:  
The Android app shows a complete different file that has previously been edited.  
Workaround: Force close the Android app.

[Pixel4a_Android-13_non-empty-new-file.mp4](https://user-images.githubusercontent.com/3355089/236349971-98d4eaa9-a04f-49b5-a473-a0ad7682aa9c.mp4)

Maybe related: #4112 _Text 3.7.2: Files mixed or emptied_

**Comment by juliushaertl** [comment]
> Here are some videos showing a wide range of problems. Including the bug which deletes all text following the cursor.

Thanks a lot for that, sharing those insights and the videos are highly appreciated.

In order to address those properly I'd split them into individual tickets.


> The last video shows an additional problem I discovered: The Android app shows a complete different file that has previously been edited. Workaround: Force close the Android app.

This looks like #4112 and I just a possible fix in #4124
 
> What about my problems and questions? (aside from the hard bugs) Can the user position indicator feature be disabled or the 30 seconds timeout be changed? What to do about indicators overlapping text I'd like to read?

Currently not but I agree that this makes sense to have so I filed https://github.com/nextcloud/text/issues/4126 for that

>  FP3_Android-10_eOS-1.4.mp4 

This looks like https://github.com/nextcloud/text/issues/1927

>  Pixel4a_Android-13.mp4 

The issue with multiple cursors has been addressed with #4053 

>  Pixel4a_Android-13_non-empty-new-file.mp4

As mentioned above this is likely resolved by #4124

I'd close this issue in favour of the individual ones. If you find further issues feel free to create new ones, ideally already separated in one github ticket per issue :)

**Comment by kolAflash** [comment]
@juliushaertl  
Thanks a lot!  
Also for splitting up everything. (I'll try to do that next time myself)  

This broke a lot of my workflows.  
I'm really relying on Nextcloud and the text app.  
I've no use for software I can't rely on.  
And I guess for many other people it's the same.  
**Would be very great if there will be an update soon!**
And again thanks a lot for your work and responsiveness!

&nbsp;

I've just applied this **WORKAROUND**:
1. Download [version for Nextcloud 25.0.6](https://github.com/nextcloud/text/releases/tag/v25.0.6), containing the text app version 3.6.0.
2. Extract and copy these files into a new folder called `text`: `appinfo babel.config.js CHANGELOG.md composer.json composer.lock COPYING css cypress.config.js img index.html jest-raw-loader.js js jsconfig.json krankerl.toml l10n lib templates vite.config.js`
3. Inside the `text` folder edit `appinfo/info.xml` and set `max-version="26"`.
4. Make a backup of the `apps/text/` folder on your Nextcloud server and replace it with the newly created `text` folder.
5. Call the Nextcloud webinterface as admin to run database migrations. (or run `occ upgrade` as www user)

&nbsp;

P.S.

I just found another bug. Sometimes when opening a definetly non empty TXT file the editor asks to use the current or the saved version. But choosing either answer gives me an empty file.  
**But I'll first see if the pending issues / merges might already fix this.**
